### PR TITLE
post: cli v9

### PR DESCRIPTION
**Pull request by cexbrayat**
TODO:

- [x] change date
- [x] update description
- [x] i18n link
- [ ] did we land `ng add` for speculoos and valdemort

### CODE REVIEW

File: _posts/2019-11-06-angular-cli-9.0.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+to specify the hosts allowed to access your application.
+
+
+## i18n
+
+TODO: localize support when it lands
+
+When building an application with i18n,
+the the document locale is now set using the `lang` attribute on `html` root element.
+For example, if you build your application  in french, then the `index.html` file contains `<html lang='fr'>`.
```
Comment (jnizet) [inline _posts/2019-11-06-angular-cli-9.0.md]: french -> French

File: _posts/2019-11-06-angular-cli-9.0.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+Until now, you could do by hand, but it was cumbersome.
```
Comment (jnizet) [inline _posts/2019-11-06-angular-cli-9.0.md]: you could do by hand -> you could of course do it by hand

File: _posts/2019-11-06-angular-cli-9.0.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+and the CLI flag `enableIvy` is now `true` by default!
+You can still switch it back to `false` if you encounter an issue.
+
+You'll notice that the first build/serve of your application is slightly longer:
+this is because `ngcc` needs to compile your dependencies.
+Once this is done, the dependencies are "Ivy compatible",
+and the following builds don't need to run `ngcc` again
+(except if you add a new dependency for example).
+This should be mostly transparent,
+as the Angular team did its best to support the most often used libraries.
```
Comment (jnizet) [inline _posts/2019-11-06-angular-cli-9.0.md]: the popular libraries

File: _posts/2019-11-06-angular-cli-9.0.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+This is probably the most awaited feature in the Angular community:
+[Ivy](/2019/05/07/what-is-angular-ivy/) is now production ready,
+and the CLI flag `enableIvy` is now `true` by default!
+You can still switch it back to `false` if you encounter an issue.
+
+You'll notice that the first build/serve of your application is slightly longer:
+this is because `ngcc` needs to compile your dependencies.
+Once this is done, the dependencies are "Ivy compatible",
+and the following builds don't need to run `ngcc` again
```
Comment (jnizet) [inline _posts/2019-11-06-angular-cli-9.0.md]: Do you know where ngcc stores the "ivyfied" libraries? Is it in node_modules, along with the original library?

File: _posts/2019-11-06-angular-cli-9.0.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+(except if you add a new dependency for example).
+This should be mostly transparent,
+as the Angular team did its best to support the most often used libraries.
+If you encounter an issue with one of your dependencies though,
+open an issue on the Angular repository,
+and someone will take a look.
+
+The build and serve command are now in AoT by default,
```
Comment (jnizet) [inline _posts/2019-11-06-angular-cli-9.0.md]: command -> commands
are now in AoT -> now use the AoT compiler

File: _posts/2019-11-06-angular-cli-9.0.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+### ng generate guard
+
+You can now specify that you want a `CanDeactivate` guard
+in addition to the already supported `CanActivate`, `CanActivateChild`, and `CanLoad`.
+I realized it was missing when working on an application for a customer,
+so I added it 🤓.
+
+### ng serve
+
+The `serve` command now accepts an `--allowed-hosts` flag,
```
Comment (jnizet) [inline _posts/2019-11-06-angular-cli-9.0.md]: an -> a

File: _posts/2019-11-06-angular-cli-9.0.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+to specify the hosts allowed to access your application.
+
+### ng build
+
+The `build` command now has an `--experimental-rollup-pass` flag.
+The name is explicit enough: this is an experiment!
+The build then leverages [Rollup](https://rollupjs.org) to, hopefully,
```
Comment (jnizet) [inline _posts/2019-11-06-angular-cli-9.0.md]: I'd remove `then`

File: _posts/2019-11-06-angular-cli-9.0.md [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+
+### ng update
+
+It is now possible to ask the CLI to create one commit per migration done
+by using `ng update --create-commits`.
+
+
+## i18n - internationalization support with @angular/localize
+
+As Angular v9 brings a re-written support for i18n, the CLI has also some changes.
```
Comment (jnizet) [inline _posts/2019-11-06-angular-cli-9.0.md]: has also some changes -> also includes some i18n-related changes

File: _posts/2019-11-06-angular-cli-9.0.md (hunk 4 above)
Comment (cexbrayat) [reply _posts/2019-11-06-angular-cli-9.0.md]: Yes. Right now, it is inside the dependency node module, in a `__ivy_ngcc__` folder, containing all the formats transformed (es5, fesm5, es2015, etc...). The package.json is then updated to reference these tranformed bundles (either replaced in place or in a dedicated field according to a `create-entry-point` option). There are some talks to put them in a cached directory, independent from node_modules, but I think it has been post-poned post v9.

**Pull request merged**